spec/controllers/tasks_controller_spec.rb

```diff
@@ -0,0 +1,289 @@
+
+        it { is_expected.to have_http_status(200) }
+        it { is_expected.to render_template(:edit) }
+
+        it 'assigns tasks with provided id to instance variable' do
+          subject
+          expect(controller.instance_variable_get(:@task)).to eq(task1)
+        end
+      end
+
+      context 'when user does not have task with provided id' do
+        let(:task_id) { task2.id }
+
+        it { is_expected.to have_http_status(400) }
+
+        it 'returns json body' do
+          subject
+
+          expect(JSON.parse(response.body)).to eql(error_response)
+        end
+      end
+    end
+
+    context 'when user is not authenticated' do
+      let(:task_id) { 0 }
+
+      it { is_expected.to have_http_status(302) }
+
+      it 'redirects to sign_in' do
+        subject
+
+        expect(response.status).to redirect_to('/users/sign_in')
+      end
+    end
+  end
+
+  describe 'PUT#update' do
+    subject { put(:update, id: task_id, task: task_params) }
+
+    context 'when user is authenticated' do
+      let(:user) { create(:user) }
+      let(:task1) { create(:task, user_id: user.id) }
+      let(:task2) { create(:task) }
+      let(:task_params) { FactoryGirl.attributes_for(:task) }
+
+      before { sign_in(user) }
+
+      context 'when user has task with provided id' do
+        let(:task_id) { task1.id }
+
+        it { is_expected.to have_http_status(302) }
+        it { is_expected.to redirect_to(task_path(id: task_id)) }
+
+        it 'updates task with provided params' do
+          subject
+
+          task = Task.find(task1.id)
+          expect(task.title).to eq(task_params[:title])
+          expect(task.body).to eq(task_params[:body])
+        end
+      end
+
+      context 'when user does not have task with provided id' do
+        let(:task_id) { task2.id }
+
+        it { is_expected.to have_http_status(400) }
+
+        it 'returns json body' do
+          subject
+
+          expect(JSON.parse(response.body)).to eql(error_response)
+        end
+      end
+    end
+
+    context 'when user is not authenticated' do
+      let(:task_params) { FactoryGirl.attributes_for(:task) }
+      let(:task_id) { 0 }
+
+      it 'redirects to sign_in' do
+        subject
+
+        expect(response).to redirect_to('/users/sign_in')
+      end
+
+      it { is_expected.to have_http_status(302) }
+    end
+  end
+
+  describe 'DELETE#destroy' do
+    subject { delete(:destroy, id: task_id) }
+
+    context 'when user is authenticated' do
+      let(:user) { create(:user) }
+      let(:task1) { create(:task, user_id: user.id) }
+      let(:task2) { create(:task) }
+
+      before { sign_in(user) }
+
+      context 'when user has task with provided id' do
+        let(:task_id) { task1.id }
+
+        it { is_expected.to have_http_status(302) }
+        it { is_expected.to redirect_to(root_path) }
+
+        it 'destroys a task' do
+          task1
+          expect { subject }.to change { Task.count }.from(1).to(0)
```

Comment:
and this is a better usage of `change from-to`
(you can leave it as it is, just commenting)